src/psyclone/transformations.py

```diff
@@ -700,22 +724,28 @@ def _directive(self, parent, children, collapse=None):
         Must be implemented by sub-class.
 
         :param parent: the parent of this Directive in the Schedule.
-        :type parent: :py:class:`psyclone.psyGen.Node`.
+        :type parent: :py:class:`psyclone.psyGen.Node`
         :param children: list of nodes that will be children of this Directive.
-        :type children: list of :py:class:`psyclone.psyGen.Node`.
+        :type children: list of :py:class:`psyclone.psyGen.Node`
         :param int collapse: the number of tightly-nested loops to which \
                              this directive applies or None.
+
         :returns: the new Directive node.
         :rtype: sub-class of :py:class:`psyclone.psyGen.Directive`.
         '''
 
-    def _validate(self, node, collapse=None):
+    def validate(self, node, options=None):
         '''
         Perform validation checks before applying the transformation
 
         :param node: the node we are checking.
-        :type node: :py:class:`psyclone.psyGen.Node`.
+        :type node: :py:class:`psyclone.psyGen.Node`
         :param int collapse: number of nested loops to collapse or None.
+        :param options: a dictionary with options for transformations.\
+                        This transform supports "collapse", which is the\
```

Comment:
Please document `options["collapse"]` as a parameter.

Reply:
Done (gee, the emails don't always specify the line numbers, which then makes it a pain to find all the locations - but I think I got them all).